src/dialect/dialect.ts

```diff
@@ -19,4 +26,56 @@ export interface Dialect extends KyselyDialect {
     callback?: ColumnBuilderCallback
   ): CreateTableBuilder<TB>;
 
+  /**
+   * This is a helper method to add a column with foreign key constraints.
+   * This is primarily useful because the `mySQL` dialect adds the constraints in a different way than `sqlite` and `postgres`.
+   *
+   * @param builder the CreateTableBuilder to add the column to.
+   * @param tableName the name of the table to add the column to.
+   * @param columnName the name of the column to add.
+   * @param targetType the type of the column to add.
+   * @param referenceTable the foreign table to reference.
+   * @param referenceColumnName the foreign column to reference.
+   * @param onDeleteAction the action to take when the referenced row is deleted.
+   *
+   * The ForeignKey name it creates in `mysql` will be in the following format:
+   * `${referenceTable}${referenceColumnName}_${tableName}${columnName}`
+   * ex: if the reference table is `users` and the reference column is `id` and the table is `profiles` and the column is `userId`,
+   * the resulting name for the foreign key is: `usersid_profilesuserId`
+   *
+   * @returns {CreateTableBuilder} the CreateTableBuilder with the added column.
+   */
+  addReferencedColumn<TB extends string>(
+    builder: CreateTableBuilder<TB & string>,
+    tableName: TB,
+    columnName: string,
+    targetType: ColumnDataType,
+    referenceTable: string,
+    referenceColumnName: string,
+    onDeleteAction: 'cascade' | 'no action' | 'restrict' | 'set null' | 'set default',
+  ): CreateTableBuilder<TB & string>;
+
+  /**
+   * This is a helper method to return an `insertId` for each dialect.
+   * We need this method because Postgres does not return an `insertId` with insert an insert.
+   *
+   * Since `returning` is supported on both `sqlite` and `postgres`, it will be used on those and ignored on `mysql`.
+   *
+   * @param db the Kysely DB object or a DB Transaction.
+   * @param table the table to insert into.
+   * @param values the values to insert.
+   * @param returning a string representing the generated key you'd like returned as an insertId.
+   *
+   *  NOTE: the `returning` value must be formatted to return an insertId value.
+   *  ex. if the generated key is `id` the string should be `id as insertId`.
+   *      if the generated key is `watermark` the string should be `watermark as insertId`.
+   *
+   * @returns {InsertQueryBuilder} object to further modify the query or execute it.
+   */
+  insertIntoReturning<DB, TB extends keyof DB = keyof DB, SE extends SelectExpression<DB, TB & string> = any>(
+    db: Transaction<DB> | Kysely<DB>,
+    table: TB & string,
+    values: InsertObject<DB, TB & string>,
+    returning: SE & `${string} as insertId`,
+  ): InsertQueryBuilder<DB, TB & string, Selection<DB, TB & string, SE & `${string} as insertId`>>;
```

Comment:
I wasn't sure on if I wanted to go this far w/ the generics... I think you know that I normally don't hah.

However in this case I really wanted to try and get the most out of making sure columns, tables, etc were respected as well as forcing the `as insertId` to avoid any accidental errors.

Figured this was a write once and it works type of thing, but let me know if you think we should do away with the generics here.

Reply:
Let's assume this is a write once forget it type of thing.